### PR TITLE
Reworked roles and permissions

### PR DESCRIPTION
**Pull request by Cerfoglg**
Now include specific permissions per projects, specifically the ability to edit and add clusters, services and subs. Also reworked the way permissions are handled.

### CODE REVIEW

File: app/controllers/v1/clusters_controller.rb [hunk 1]
```diff
@@ -116,33 +117,48 @@ def self.add_common_params(api)
 
   # GET /projects/:project_id/clusters
   def index
-    authorize @project
+    if @role.nil? and !current_actor.superadmin
```
Comment (taliaga) [inline app/controllers/v1/clusters_controller.rb]: Side note, I've just realised the string distance between superadmin and superman is smaller than I'd've thought.

File: spec/requests/v1/role_levels_spec.rb [hunk 2]
```diff
@@ -69,10 +73,12 @@
...
   # Test suite for PUT /role_levels/:id
   describe 'PUT /v1/role_levels/:id' do
-    let(:valid_attributes) { { name: 'User' } }
+    let(:valid_attributes) do
+      { name: 'User'}.to_json
+    end
 
     context 'when the record exists' do
-      before { put "/v1/role_levels/#{role_level_id}", params: valid_attributes }
+      before { put "/v1/role_levels/#{role_level_id}", params: valid_attributes, headers: headers }
```
Comment (taliaga) [inline spec/requests/v1/role_levels_spec.rb]: May be worth (for all these tests) to test one withouth headers to get a 403. Wdyt?

Comment (taliaga) [reply spec/requests/v1/role_levels_spec.rb]: and/or some "read-only" vs "edit" variants

Comment (Cerfoglg) [reply spec/requests/v1/role_levels_spec.rb]: @taliaga Before merging this I'll update the test to be a bit more robust, it's probably for the best in the long run.

**Pull request merged**